### PR TITLE
Add docker app specific annotations on push

### PR DESCRIPTION
**Pull request by rumpl**
**- What I did**

[This PR](https://github.com/docker/cnab-to-oci/issues/40) will remove the docker app specific annotations on push, this PR adds them.

**- How I did it**

cnab-to-oci has a callback for users to make last-minute changes to the index, I added a function that adds our annotations

**- How to verify it**

`¯\_(ツ)_/¯ `

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

File: internal/commands/push.go [hunk 1]
```diff
@@ -21,10 +21,28 @@ import (
 	"github.com/docker/docker/pkg/term"
 	"github.com/docker/docker/registry"
 	"github.com/morikuni/aec"
+	ocischemav1 "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
+// Type aliases to clarify to which annotation the values belong
+type dockerAppFormatValue = string
```
Comment (silvin-lubecki) [inline internal/commands/push.go]: Can you try this without the type aliasing? Just standard types?

File: internal/commands/push.go [hunk 2]
```diff
@@ -114,14 +132,20 @@ func runPush(dockerCli command.Cli, name string, opts pushOptions) error {
...
-	descriptor, err := remotes.Push(context.Background(), bndl, retag.cnabRef, resolverConfig.Resolver, true)
+	descriptor, err := remotes.Push(context.Background(), bndl, retag.cnabRef, resolverConfig.Resolver, true, addAnnotations)
 	if err != nil {
 		return errors.Wrapf(err, "pushing to %q", retag.cnabRef)
 	}
 	fmt.Fprintf(os.Stdout, "Successfully pushed bundle to %s. Digest is %s.\n", retag.cnabRef.String(), descriptor.Digest)
 	return nil
 }
 
+func addAnnotations(index *ocischemav1.Index) error {
```
Comment (silvin-lubecki) [inline internal/commands/push.go]: Don't return an error here?

File: internal/commands/push.go [hunk 3]
```diff
@@ -114,14 +132,20 @@ func runPush(dockerCli command.Cli, name string, opts pushOptions) error {
...
+	index.Annotations[DockerAppFormatAnnotation] = DockerAppFormatCNAB
```
Comment (silvin-lubecki) [inline internal/commands/push.go]: nit: Check if the `Annotations` map is not nil?

File: internal/commands/push.go (hunk 2 above)
Comment (silvin-lubecki) [inline internal/commands/push.go]: Rename to `withAppAnnotations` ?

File: internal/commands/push.go (hunk 1 above)
Comment (rumpl) [reply internal/commands/push.go]: Done

File: internal/commands/push.go (hunk 2 above)
Comment (rumpl) [reply internal/commands/push.go]: Done

File: internal/commands/push.go (hunk 3 above)
Comment (rumpl) [reply internal/commands/push.go]: Aaaand done

File: internal/commands/push.go [hunk 4]
```diff
@@ -114,14 +127,22 @@ func runPush(dockerCli command.Cli, name string, opts pushOptions) error {
...
+	descriptor, err := remotes.Push(context.Background(), bndl, retag.cnabRef, resolverConfig.Resolver, true, withAppAnnotations)
 	if err != nil {
 		return errors.Wrapf(err, "pushing to %q", retag.cnabRef)
 	}
 	fmt.Fprintf(os.Stdout, "Successfully pushed bundle to %s. Digest is %s.\n", retag.cnabRef.String(), descriptor.Digest)
 	return nil
 }
 
+func withAppAnnotations(index *ocischemav1.Index) error {
+	if index.Annotations != nil {
```
[the remark this answers is not in the document]
Comment (rumpl) [reply internal/commands/push.go]: That's what's going on no?

**Pull request merged**